server/autolinkplugin/plugin.go

```diff
@@ -61,7 +62,26 @@ func contains(team string, channel string, list []string) bool {
 	return false
 }
 
+func (p *Plugin) isBotUser(userId string) bool {
+	user, appErr := p.API.GetUser(userId)
+	if appErr != nil {
+		// NOTE: Not sure how we want to handle errors here, we can either:
+		// * assume that occasional rewrites of Bot messges are ok
+		// * assume that occasional not rewriting of all messages is ok
+		// Let's assume for now that former is a lesser evil
+		mlog.Error("failed to check if message for rewriting was send by a bot", zap.Error(appErr))
+		return false
+	}
+
+	return user.IsBot
+}
+
 func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post, string) {
+	if p.isBotUser(post.UserId) {
+		mlog.Debug("not rewriting message from bot", zap.String("userId", post.UserId))
```

Comment:
In the future, the whole code base should use [`p.API.LogDebug`](https://developers.mattermost.com/extend/plugins/server/reference/#API.LogDebug) instead. It would be nice if you start using the with this PR. But it's not blocking.